### ISSUE TITLE
SC2151: do not warn on `--` argument to `return`

### ISSUE THREAD

**Issue by mooreye** [koalaman/shellcheck#3403]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2151
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

func() { return -- "$1"; }
```

#### Here's what shellcheck currently says:

One can have such function in their `~/.bashrc` where `--` prevents injection like `--help` and ensures that anything is passed as return status to `return`, even if it's wrong (not a number).

#### Here's what I wanted or expected to see:

Make an exception so that `return` does not trigger SC2151 warning if it has `--` as first argument, and then some other argument.
